modules/cli/cli.go

```diff
@@ -119,29 +119,45 @@ type flagsForFurtherEvaluation struct {
 }
 
 const (
-	envVarFlagPrefix = "GATEWAY_OPERATOR_"
+	envDeprecatedVarFlagPrefix = "GATEWAY_OPERATOR_"
+	envVarFlagPrefix           = "KONG_OPERATOR_"
 )
 
+// setFlagFromEnvVar looks for the env var name corresponding to `f`: if the env val is found its value is set
+// to the flag. In case of error it panics.
+func setFlagFromEnvVar(f *flag.Flag) {
+	envKey := fmt.Sprintf("%s%s", envVarFlagPrefix, strings.ToUpper(strings.ReplaceAll(f.Name, "-", "_")))
+	if envValue, envSet := os.LookupEnv(envKey); envSet {
+		if err := f.Value.Set(envValue); err != nil {
+			panic(fmt.Errorf("environment binding failed for variable %s: %w", envKey, err))
+		}
+		// ignore deprecated env var if newer is set.
+		return
+	}
+
+	deprecatedEnvKey := fmt.Sprintf("%s%s", envDeprecatedVarFlagPrefix, strings.ToUpper(strings.ReplaceAll(f.Name, "-", "_")))
+
+	if envValue, envSet := os.LookupEnv(deprecatedEnvKey); envSet {
+		fmt.Printf("WARN: %q env variable is deprecated, please use %q instead\n", deprecatedEnvKey, envKey)
+		// move to envKey var as that is what we look at in case of Set error.
+		envKey = deprecatedEnvKey
+		if err := f.Value.Set(envValue); err != nil {
+			panic(fmt.Errorf("environment binding failed for variable %s: %w", envKey, err))
+		}
+	}
```

Comment:
Nit: I'd suggest to consider making [the "happy path" the least indented ](https://maelvls.dev/go-happy-line-of-sight/)

Something along the lines of: (might not compile)

```suggestion
	deprecatedEnvKey := fmt.Sprintf("%s%s", envDeprecatedVarFlagPrefix, strings.ToUpper(strings.ReplaceAll(f.Name, "-", "_")))
	if envValue, envSet := os.LookupEnv(deprecatedEnvKey); envSet {
		fmt.Printf("WARN: %q env variable is deprecated, please use %q instead\n", deprecatedEnvKey, envKey)
		// move to envKey var as that is what we look at in case of Set error.
		if err := f.Value.Set(envValue); err != nil {
			panic(fmt.Errorf("environment binding failed for variable %s: %w", deprecatedEnvKey, err))
		}
	}
	if envValue, envSet := os.LookupEnv(envKey); envSet {
		if err := f.Value.Set(envValue); err != nil {
			panic(fmt.Errorf("environment binding failed for variable %s: %w", envKey, err))
		}
	}
```

WDYT?

Reply:
uh, right, for sure we don't need anymore the `envKey = deprecatedEnvKey` ti should be dropped... good catch!
What I like from your proposal is that in case we have both the deprecated and the new variable we still print the WARN message.